### DISCUSSION TITLE
is there a way I can get the password from the connection object from objection explorer ?

### DISCUSSION THREAD

**Discussion by rajsinga** [microsoft/azuredatastudio#16902] [General]
I am trying something like this, I have an active connection to oracle server, I need to get the connection info(along with the password) from a plugin. but password is always nul string, any ideas please ?

azdata.connection.getCurrentConnection().then(connection => {
            let connectionId = connection ? connection.connectionId : 'No connection found!';
            vscode.window.showInformationMessage(connection.userName);
            vscode.window.showInformationMessage(connection.password);
            azdata.connection.getConnectionString(connectionId, true).then(connectionStr => {
                vscode.window.showInformationMessage(connectionStr);
            });
        }, error => {
             console.info(error);
        });

**Comment by Charles-Gagnon** [comment]
Have you tried using [getCredentials](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/azdata.d.ts#L144)? That should return an object with a `password` property containing the password.

**Reply by rajsinga** [reply]
thanks charles, I used it and it worked,

**Reply by rajsinga** [reply]
tried something like this, it worked.
            azdata.connection.getCredentials(connectionId).then(creds => {
                logger.logError(creds['password']);
            });
